### ISSUE TITLE
[a11y]: loading and inline loading not consistently providing status programmatically

### ISSUE THREAD

**Issue by mbgower** [carbon-design-system/carbon#11259]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

https://react.carbondesignsystem.com/?path=/story/components-inlineloading--inline-loading

### Automated testing tool and ruleset

n/a

### Assistive technology

JAWS, keyboard

### Description

There are a number of considerations around loading patterns in Carbon that need to be addressed, ranging from clear accessibility bugs to inconsistencies. Several are listed here; I will add a few more after reviewing with CAG.

### Default inline loading
The [default implementation of inline loading](https://react.carbondesignsystem.com/?path=/story/components-inlineloading--inline-loading) on storybook uses aria-live to surface status (good!); however, the implementation is sometimes causing overly verbose output from the JAWS screen reader (not good). Note that any changes to the text can get read 3 times! (Haven't been able to consistently replicate this problem.) This may be partly a result of the storybook container; testing needs to be carried out on an actual implementation
![image](https://user-images.githubusercontent.com/14298245/164316694-a6609b30-9cc0-4acd-a2d6-5d8edf7f02ff.png)

   1. part of this may be due to aria-live being set to assertive (instead of polite) OR
   3. by aria-live assigned to multiple elements (the div and its child), or
   4. by having the child attribute assigned both a live region and what I suspect is an unnecessary label ("Active loading indicator") or
   5. some combination of these

I recommend considering a simple label for the symbol (for instance "loading") that can provide sufficient information whether the "Loading data..." text is also displayed or not.

### UX example inline loading
The second working example "Ux Example" is an interaction comprising a set of Cancel and Submit buttons. 
![image.png](https://zenhub.ibm.com/images/5cab73f442b8dc5ede7b84cc/1b816b0c-326a-4f38-a137-57e32a3639a6)
When you activate Submit, the text changes to "Submitting..." and then "Submitted" before returning to "Submit"
This example does not have an aria-live region set and so is not announced by screen readers. ([sev 2 BUG](https://ibm.box.com/s/thsragowvz98nlviduxlfktyso4iext3)) This is easily solved by putting aria-live on the div preceding the button --[JAWS announces the text changes](https://ibm.box.com/s/itq68565uz796mp1jgyzxdsn124ay93u). Without the storybook, I think the updates would read fine (although the visual 'pause' between "Submitted" and the text returning to "Submit" is lost on a screen reader. But I would do some testing in a real world environment.

![image.png](https://zenhub.ibm.com/images/5cab73f442b8dc5ede7b84cc/76fefddc-5ba4-48de-884a-2aaca4901e5f)

### Loading Default example
The main difference between Loading and Inline loading appears to be that Loading is thought of as strictly a visual animation of a spinning wheel, while inline loading can have text, etc

The loading icon is announced okay by JAWS. It reads out the region, the image and "indicator" (which I think is the result of a buffer and the aria-live assertive setting).

> Active loading indicator
Active loading indicator graphic
indicator

 I find this redundant, and don't understand why the choice was made to surface it all. It would be worth a bit of investigation and should be easily solved, but it is a fairly trivial repetition, so I'm not even sure I'd describe it as a bug, more of a non-delightful experience.

### Why are inline loading and loading treated as separate components?

Inline loading is a separate component from Loading on the Carbon Design System website and storybook; however in Sketch and Figma, inline loading is a variant on Loading.  I think this makes sense, overall, and I suggest the two components be combined. If desired to keep separate, then they should be separated in the Design kit, IMO.

### Sketch vs Figma vs implementation
Note that there _are_ some differences in how Figma and Sketch show the inline loading that should be addressed. Sketch includes inline loading on text inputs; Figma does not. Figma includes a icon-less version (which I think is the 'inactive' one); Sketch does not. Neither show the above implemented variant of a button that updates its text. I consider that almost a different variant.
<img width="866" alt="image" src="https://user-images.githubusercontent.com/14298245/164477377-3562b463-24a6-48d4-9c7e-5cf72ecf218b.png">

![image](https://user-images.githubusercontent.com/14298245/164326328-7bab826b-e88d-4721-b623-00ef780fa9ec.png)
![image](https://user-images.githubusercontent.com/14298245/164326397-f1db984c-8a8e-41de-be98-955efdc89b7c.png)

### Purpose of inactive?
Speaking of inactive, I'm trying to understand the use case for this. How do you have an inactive upload? There are programmatic advantages to having the inline loading established programmatically (since you can set up the live region); however, I don't understand inactive's usage from a documentation/designer perspective.
### WCAG 2.1 Violation

https://www.w3.org/WAI/WCAG21/Understanding/status-messages.html

### CodeSandbox example

https://react.carbondesignsystem.com/?path=/story/components-inlineloading--inline-loading

### Steps to reproduce

1. Go to https://react.carbondesignsystem.com/?path=/story/components-inlineloading--inline-loading
1. Turn on JAWS
2. Listen to output when loading page, or modifying the values

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by mbgower** [comment]
Looking at this a bit more, I think we want to have a scalable implementation.
`<div id="root"><div class="cds--inline-loading" aria-live="assertive"><div class="cds--inline-loading__animation"><div aria-atomic="true" aria-labelledby="loading-id-1" aria-live="assertive" class="cds--loading cds--loading--small"><label id="loading-id-1" class="cds--visually-hidden">Active loading indicator</label><svg class="cds--loading__svg" viewBox="0 0 100 100"><title>Active loading indicator</title><circle class="cds--loading__background" cx="50%" cy="50%" r="44"></circle><circle class="cds--loading__stroke" cx="50%" cy="50%" r="44"></circle></svg></div></div><div class="cds--inline-loading__text">Loading data...</div></div></div>`

 I suggest investigating this approach:

1. Remove the label; it adds no value and makes it worse for some
`<div id="root"><div class="cds--inline-loading" aria-live="assertive"><div class="cds--inline-loading__animation"><div aria-atomic="true" aria-labelledby="loading-id-1" aria-live="assertive" class="cds--loading cds--loading--small"><svg class="cds--loading__svg" viewBox="0 0 100 100"><title>Active loading indicator</title><circle class="cds--loading__background" cx="50%" cy="50%" r="44"></circle><circle class="cds--loading__stroke" cx="50%" cy="50%" r="44"></circle></svg></div></div><div class="cds--inline-loading__text">Loading data...</div></div></div>`

2. Make the svg title element text reflect the state. Keep it succinct ("loading", "finished", "error") instead of wordy ("Active loading indicator"). The svg title is picked up in the accessibility tree as the name.
`<svg class="cds--loading__svg" viewBox="0 0 100 100"><title>Loading</title><circle class="cds--loading__background" cx="50%" cy="50%" r="44"></circle><circle class="cds--loading__stroke" cx="50%" cy="50%" r="44"></circle></svg>`

3. When a text label is provided, use it in place of the svg title text (or possibly as a concatenation).
So, if the shown text was "Loading data...", the user would hear that. Here, I just stole the id already in the code...
` <div id="root"><div class="cds--inline-loading" aria-live="assertive"><div class="cds--inline-loading__animation"><div aria-atomic="true" aria-labelledby="loading-id-1" aria-live="assertive" class="cds--loading cds--loading--small"><label class="cds--visually-hidden">Loading</label><svg class="cds--loading__svg" viewBox="0 0 100 100"><title>Loading</title><circle class="cds--loading__background" cx="50%" cy="50%" r="44"></circle><circle class="cds--loading__stroke" cx="50%" cy="50%" r="44"></circle></svg></div></div><div id="loading-id-1"  class="cds--inline-loading__text">Loading data...</div></div></div>`

Where there is no text supplied, you would need to strip out the id to ensure you retain the svg title as the name. Actually, another way you may be able to do it is to ensure that the visible text is inside the div with the aria-live? Without an implemented example, it's hard to test. Storybook really doesn't give me enough in this sample.

**Comment by mbgower** [comment]
@tay1orjones any updates?

**Comment by tay1orjones** [comment]
@mbgower thanks for writing all this up! Your suggestions sound great to me and we can investigate that approach. We marked this as a sev 2 since it's technically still reading, just a bit more verbose than preferred.

**Comment by tw15egan** [comment]
Dev fixes have been implemented but leaving open for other design considerations 

https://github.com/carbon-design-system/carbon/pull/11759#issuecomment-1183238533

**Comment by tay1orjones** [comment]
I made a new issue for the other considerations so we can close this https://github.com/carbon-design-system/carbon/issues/11821
